test/cp.js

```diff
@@ -451,6 +451,45 @@ test('-R implies -P', t => {
   });
 });
 
+test('-Ru respects the -u flag recursively (don\'t update newer file)', t => {
+  const dir = `${t.context.tmp}/cp-Ru`;
+  const sourceDir = `${dir}/old`;
+  const sourceFile = `${sourceDir}/file`;
+  const destDir = `${dir}/new`;
+  const destFile = `${destDir}/file`;
+  [sourceDir, destDir].forEach(d => shell.mkdir('-p', d));
+  shell.ShellString('foo\n').to(sourceFile);
+  shell.ShellString('bar\n').to(destFile);
+  // Get the old mtime for dest
+  const oldTime = fs.statSync(destFile).mtimeMs;
+  // Send the old file to two days ago
+  shell.touch('-m', oldTime - (2 * 24 * 60 * 60 * 1000), sourceFile);
+  // Now, copy the old dir to the new one
+  shell.cp('-Ru', sourceDir, destDir);
+  // Check that dest has not been updated
+  t.is(shell.cat(destFile).stdout, 'bar\n');
+});
+
+test('-Ru respects the -u flag recursively (update older file)', t => {
+  const dir = `${t.context.tmp}/cp-Ru`;
+  const sourceDir = `${dir}/old`;
+  const sourceFile = `${sourceDir}/file`;
+  const destDir = `${dir}/new`;
+  const destFile = `${destDir}/file`;
+  [sourceDir, destDir].forEach(d => shell.mkdir('-p', d));
+  shell.ShellString('foo\n').to(sourceFile);
+  shell.ShellString('bar\n').to(destFile);
+  // Get the old mtime for dest
+  const oldTime = fs.statSync(destFile).mtimeMs;
+  // Send the source file to two days ahead
+  shell.touch('-m', oldTime + (2 * 24 * 60 * 60 * 1000), sourceFile);
+  // Now, copy the old dir to the new one
+  shell.cp('-Ru', sourceDir, destDir);
+  // Get the new mtime for dest
```

Comment:
nit: I think this comment is leftover from a previous commit (you're not checking mtimes anymore)